### PR TITLE
Feature/authentication

### LINKED ISSUE #32: restrict appview before login

### PR DESCRIPTION
**Pull request by cassiebaer**
closes #32, #42, #37 

Facebook auth needs to be fixed. We'll prob. need a join table as discussed early on in the proj.

### CODE REVIEW

File: server/models/User.js [hunk 1]
```diff
@@ -28,11 +28,21 @@ var User = db.define('user', {
     type: Sequelize.STRING,
     allowNull: true
   }
+}, {
+  instanceMethods: {
+    verifyPassword: function(password) {
+      return pCompare(password, this.password);
+    }
+  }
 });
```
Comment (danejordan) [inline server/models/User.js:37]: huzzah looks like this decided to work this time! :P

**Comment by Dane456** [comment]
🚢

**Comment by cassiebaer** [comment]
Don't merge yet, still working on register

**Comment by cassiebaer** [comment]
Somebody may want to prettify the forms, but auth should be working now

File: server/alphadeltaninerniner.service.js [hunk 2]
```diff
@@ -2,26 +2,32 @@ let http = require('http');
...
-function decorate(app) {
+function decorate(app, session) {
   let server = http.Server(app);
   let io = socketIo(server);
+  io.use((socket, next) => {
+    session(socket.request, socket.request.res, next);
+  });
   io.on('connection', socket => {
+    if (!socket.request.session.passport) {
+      socket.emit('rumi error', {message: 'Please reauthenticate'});
```
Comment (danejordan) [inline server/alphadeltaninerniner.service.js:20]: is 'rumi error' getting listened for anywhere yet?

File: server/models/Task.js [hunk 3]
```diff
@@ -25,11 +26,16 @@ var Task = db.define('task', {
...
-        this.dueBy = new Date(this.dueBy).getTime() + this.interval;
-        this.addCompleted(completed);
-        return this.save();
+    complete: function(userId) {
+      return Completed.create().then(completed => {
+        return User.findById(userId).then(user => {
+          completed.setUser(user);
+          return completed.save();
+        }).then(completed => {
+          this.dueBy = new Date(this.dueBy).getTime() + this.interval;
```
Comment (trevorwhealy) [inline server/models/Task.js:35]: I love this, nice job

**Comment by danejordan** [comment]
🚢

**Comment by trevorwhealy** [comment]
:shipit:

**Pull request merged**